### PR TITLE
When user clicks "Previous" with invalid input, show a modal

### PR DESCRIPTION
**Pull request by ssandbekkhaug**
### Description

1. User is filling out a block
1. User has invalid input
1. User clicks "previous"
1. Page reloads via routes
1. User sees modal
1. Modal lets the user discard updates and exit, or remain on page

I acknowledge this implementation is a bit janky. I also considered using HTMX. Suggestions are welcome!

This PR adds functionality. Styling will be added in a follow-up PR after I've confirmed some details with Sijia.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

#### User visible changes

- [x] Followed steps to [internationalize](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#application-strings) any new strings
  - [x] Added context strings to new [messages](https://github.com/civiform/civiform/blob/main/server/conf/i18n/messages)
  - [x] Didn't use a message in applicant facing code that isn't translated yet (unless behind a flag)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Made equivalent changes in Thymeleaf for applicant-facing features and changes (or created an issue in the [Northstar epic](https://github.com/orgs/civiform/projects/1/views/94) to track that it's missing from Thymeleaf)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

### Instructions for manual testing

1. Enable North Star
1. Use CiviForm as user
1. Start an application to a program
1. Fill out a block with invalid input
1. Click "Previous"

### Issue(s) this completes

Part of #8065

### CODE REVIEW

**Comment by ssandbekkhaug** [comment]
* **#8511** <a href="https://app.graphite.dev/github/pr/civiform/civiform/8511?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ssandbekkhaug and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ssandbekkhaug** [comment]
### Merge activity

* **Sep 6, 3:54 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/8511) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

File: server/app/views/applicant/ModalFragment.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div
-  th:fragment="modalContainer(showProgramIndexLogin, showReviewLogin, showUpsellLogin)"
+  th:fragment="modalContainer(showProgramIndexLogin, showReviewLogin, showUpsellLogin, showValidationModal)"
```
[the remark this answers is not in the document]
Comment (ssandbekkhaug) [reply server/app/views/applicant/ModalFragment.html:2]: Good question. This is some tech debt tracked in #7880.

I went ahead and refactored North Star modals in https://github.com/civiform/civiform/pull/8560. I'll update this PR after #8560 is merged.

**Review by gwendolyngoetz — APPROVED** [review]
I'm still not fond of the sentinel element to be a boolean, but I don't have any better options right now. Long term if the uswds modals don't have a programmatic way to trigger them we should look to maintain only the styling and leverage the modern builtin browser dialog api.

**Pull request merged**